tests/integration/targets/ini_file/tasks/tests/02-values.yml

```diff
@@ -0,0 +1,820 @@
+---
+
+## testing values
+
+- name: "test-values 1 - set 'state=present' and 'values=[]' and 'allow_no_value=false' and fail"
+  ini_file:
+    path: "{{ output_file }}"
+    section: cars
+    option: audi
+    values: []
+    allow_no_value: false
+  register: result1
+  ignore_errors: true
+
+- name: test-values 1 - verify error message
+  assert:
+    that:
+      - result1 is not changed
+      - result1.msg == "Parameter 'value(s)' must be defined if state=present and allow_no_value=False."
+
+
+- name: "test-values 2 - set 'state=present' and omit 'values' and 'allow_no_value=false' and fail"
+  ini_file:
+    path: "{{ output_file }}"
+    section: cars
+    option: audi
+    allow_no_value: false
+  register: result2
+  ignore_errors: true
+
+- name: test-values 2 - verify error message
+  assert:
+    that:
+      - result2 is not changed
+      - result2.msg == "Parameter 'value(s)' must be defined if state=present and allow_no_value=False."
+
+
+- name: "test-values 3 - ensure 'fav=lemonade' and 'fav=cocktail' is 'present' in section '[drinks]' in specified file"
+  ini_file:
+    path: "{{ output_file }}"
+    section: drinks
+    option: fav
+    values:
+      - lemonade
+      - cocktail
+    state: present
+  register: result3
+
+- name: test-values 3 - read content from output file
+  slurp:
+    src: "{{ output_file }}"
+  register: output_content
+
+- name: test-values 3 - set expected content and get current ini file content
+  set_fact:
+    expected3: |
+
+      [drinks]
+      fav = lemonade
+      fav = cocktail
+    content3: "{{ output_content.content | b64decode }}"
+
+- name: test-values 3 - Verify content of ini file is as expected and ini_file 'changed' is true
+  assert:
+    that:
+      - result3 is changed
+      - result3.msg == 'section and option added'
+      - content3 == expected3
+
+
+- name: "test-values 4 - remove option 'fav=lemonade' from section '[drinks]' in specified file"
+  ini_file:
+    path: "{{ output_file }}"
+    section: drinks
+    option: fav
+    values:
+      - lemonade
+    state: absent
+  register: result4
+
+- name: test-values 4 - read content from output file
+  slurp:
+    src: "{{ output_file }}"
+  register: output_content
+
+- name: test-values 4 - set expected content and get current ini file content
+  set_fact:
+    expected4: |
+
+      [drinks]
+      fav = cocktail
```

Comment:
IMHO, this should be the result with `exclusive=false`, that is not the case here.

More precisely, the current (mainline) behavior is to remove an option when `state=absent`, whenever the `value` is defined or not, and if it is, by ignoring its actual content. This is what we should get when `exclusive=true` (the default)

Reply:
I changed the test and added `exclusive=false`.